shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java

```diff
@@ -517,8 +518,39 @@ private CharSequence getClipboardData(PlatformChannel.ClipboardContentFormat for
       if (clip == null) return null;
       if (format == null || format == PlatformChannel.ClipboardContentFormat.PLAIN_TEXT) {
         ClipData.Item item = clip.getItemAt(0);
-        if (item.getUri() != null)
-          activity.getContentResolver().openTypedAssetFileDescriptor(item.getUri(), "text/*", null);
+        // First, try getting clipboard data as text; no further processing
+        // required if so.
+        CharSequence itemText = item.getText();
+
+        if (itemText == null) {
```

Comment:
Nonblocking nit: consider extracting uri logic into a method or methods.

Reply:
Undid this due to some recent changes here that actually made the new method more confusing.